Share branch justin

jreigel opened the pull request.

jreigel committed:
Merge with Mohammeds code for master_testing

jreigel committed:
Added icons

Added the icons needed for buttons for management screens, and mainhub

jreigel committed:
Product Panel and stub

Cleaned up code in management screens by placing reused code in
productPanel and testingStub classes.

jreigel committed:
Merge branch 'master' of https://github.com/SER316GroupF/c3-entrepreneurship-simulator into share_branch_justin

jreigel committed:
Fix CurrentProducts and RetiredProduts icons

CurrentProducts previously had relaunch and retired had retire icons.
swapped these so the appropriate icons belong to the appropriate
classes.

jreigel committed:
Product interface and stub

added getters to product interface and stub for product history (number
of products sold, total income generated by product)

jreigel committed:
Product history panel

Added method to place product history in the product info panel as well.

jreigel committed:
Product History, .gitignore

Created a product history screen (front end) and implemented it properly
in the enumerated singleton and navigation panel. .gitignores were
updated, not sure why.

jreigel committed:
Merge remote-tracking branch 'origin/master_testing' into share_branch_justin

jreigel committed:
Updated council and marketing screens

Added home buttons and initialize methods(for screen transitioning)
Added these screens to the CompanyPanel and AllManagementScreens

jreigel committed:
DifficultySelectionScreen

Updated documentation (Mainhub is implemented so todo: Manhib has been
removed)

jreigel committed:
ProductDesigns

cleared popupWindow, cleared product listeners, removed defaults from
new products

jreigel committed:
Product

Updated product class to include image path.

jreigel committed:
Management Screens

Updated initialize methods for efficiency.

jreigel committed:
ManagePanel

Updated ManagePanel for aesthetic purposes. Previous managePanel would
slightly jump around the screen.

jreigel committed:
Updated MainHub

set the screens that don't exist yet to jump to null (previously was
game.getScreen() which had unexpected results) Linked Council button to
Company Panel screen.

jreigel committed:
Simulation Screen

changed createNavigationPanel to return void. (create navigation panel
method was updated to add the navigation panel to the stage)

jreigel committed:
Mass Format

Went through code and did ctrl+ shift + F and ctrl +shift+O  to format
and remove unneeded imports from various classes. Updated documentation
where appropriate.

jreigel committed:
Council Panel

Formatted and added documentation.

zcmoore commented on core/.gitignore:
```diff
@@ -0,0 +1,3 @@
+/bin
+/bin
+/bin
```
Does this need to be declared thrice?

jreigel replied:
I have no clue why or how this is happening.

jreigel replied:
Fixed.

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+import com.badlogic.gdx.scenes.scene2d.utils.ClickListener;
+
+import edu.asu.c3simulator.screens.MainHub;
+//import edu.asu.c3simulator.screens.MainHub;
```
Commented code/imports should be removed

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+import edu.asu.c3simulator.screens.MainHub;
+//import edu.asu.c3simulator.screens.MainHub;
+/**
+ * This class creates a Home button actor, places it in the bottom left of the screen. Changes the screen to MainHub when clicked.
```
Don't start with "This class"

Also, the class does not "create a Home button actor." It IS a HomeButton. The class javadoc should tell me what a HomeButton is.

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+* @author Reigel, Justin 
+* 
+*/
+public class HomeButton extends Table{
```
This looks a lot like a button that is given a specific position and listener. It doesn't seem to add any functionality or properties that are not contained in Button.

I propose that we replace this class with a method in WidgetFactory

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+	
+	private Image homeIcon;
+	public float homeLeft;
+	public float homeBottom;
```
These are public, but it doesn't look like the position is backed by these values. Are they used anywhere, or by anything else?

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+
+/**
+ * Creates a button that in turn creates a popup window when clicked. The popup
+ * window requires confirmation via button click before selling the company.
```
This sounds like a method again

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+package edu.asu.c3simulator.widgets;
```
The package is declared properly, but the file is in the wrong folder. Make sure it is in its package directory.

jreigel replied:
I double checked. This file is in edu.asu.c3simulator.widgets.

zcmoore replied:
It is a duplicate.
This file is in core/HomeButton.java

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+package edu.asu.c3simulator.widgets;
```
The package is declared properly, but the file is in the wrong folder. Make sure it is in its package directory.

jreigel replied:
I double checked. This file is in edu.asu.c3simulator.widgets.

zcmoore replied:
Duplicate. This file is in core/SellCompany.java

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+*/
+public class HomeButton extends Table{
+	
+	private Image homeIcon;
```
Does this need to be stored? It looks like it is only used once

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+		FileHandle iconLocation = Gdx.files.internal("images/home-button.png");
+		Texture iconTexture = new Texture(iconLocation);
+		this.homeIcon = new Image(iconTexture);
+		add(homeIcon).size(150);
```
This all would be perfect for a factory method

jreigel replied:
That is a good idea. I can look into it for future cleaning or refactoring. But changing this now would also require me to make changes in 12 different screens. Could be more efficient now to just keep it the way it is.

zcmoore replied:
All changes in the 12 different screens are all a single line...
change: HomeButton homeButton = new HomeButton();
to: Actor homeButton = WidgetFactory.createHomeButton();

Do you mind elaborating on what you mean by "more efficient?"

zcmoore commented on core/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+		homeLeft = this.getPrefWidth()/2;
+		homeBottom = this.getPrefHeight()/2;
+		this.setPosition(homeLeft, homeBottom);
+		this.addListener(new ClickListener(){
```
Do we have a generic TransitionListener yet?

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+ * @author Justin
+ *
+ */
+public class SellCompany extends Table
```
This isn't a very descriptive class name. Is is a panel? A button? A complex widget?

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+	private BitmapFont font;
+	private Window window;
+
+	public SellCompany(final Game game, final Skin skin)
```
Any reason you declared these final?

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+	public SellCompany(final Game game, final Skin skin)
+	{
+		super();
+		final TextButton continueButton = new TextButton("Continue", skin);
```
Requesting justification for final modifier.

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+			public void clicked(InputEvent event, float x, float y)
+			{
+				System.out.println("Continue!");
+				// TODO: Delete actual business
```
The continue button will delete the business?? Why?

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+			@Override
+			public void clicked(InputEvent event, float x, float y)
+			{
+				System.out.println("Cancel.");
```
These statements should be left out of production code unless they are placeholders. If you want notifications, use a logger, such as the ApacheLogger declared in our spec.

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+				}
+			}
+		});
+		button = new TextButton("Sell", skin);
```
This could use a more descriptive name. sellButton, perhaps

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+
+	private TextButton button;
+	private TextButtonStyle textButtonStyle;
+	private BitmapFont font;
```
I didn't see these referenced past their initial conception. Did I miss something?

zcmoore commented on core/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+				window.setSize(550f, 200f);
+				window.setColor(1f, 1f, 1f, 1f);
+				window.setResizable(false);
+				getStage().addActor(window);
```
Factory method?

zcmoore commented on core/src/edu/asu/c3simulator/screens/AllManagementScreens.java:
```diff
@@ -0,0 +1,90 @@
+ */
+public enum AllManagementScreens
+{
+	PRE_MARKET, CURRENT_PRODUCTS, RETIRED_PRODUCTS, CURRENT_PRODUCTS_GROWTH, PRODUCT_HISTORY, GROWTH_DEMAND, EMPLOYMENT, MARKETING, BUSINESS_DIRECTION, FUNDING, INDUSTRY, TASKS, COMPANY_PANEL, COMPLETED_TASKS, TASK_MANAGEMENT;
```
It looks like the formatting got thrown off.

The formatter needs to be run on this

zcmoore commented on core/src/edu/asu/c3simulator/screens/AllManagementScreens.java:
```diff
@@ -0,0 +1,90 @@
+			((SimulationScreen) COMPLETED_TASKS.instance)
+					.createNavigationPanel();
+			((SimulationScreen) TASK_MANAGEMENT.instance)
+					.createNavigationPanel();
```
This needs refactoring. Perhaps an interface along the lines of C3Screen should be used to connect these, and provide an initialize() method

malyahy2 replied:
Actually I already have an interface called "SimulationScreen" which does just that using a method called createNavigationPanel.

zcmoore replied:
Awesome! 

Derp. I just noticed the cast.... x)
In any case, let's change line 31 to SimulationScreen instead of Screen and get rid of all these casts.

zcmoore replied:
Let's turn it into a loop, too. No need to have these excess lines.

zcmoore commented on core/src/edu/asu/c3simulator/screens/AllManagementScreens.java:
```diff
@@ -0,0 +1,90 @@
+			((SimulationScreen) FUNDING.instance).createNavigationPanel();
+			((SimulationScreen) INDUSTRY.instance).createNavigationPanel();
+			((SimulationScreen) TASKS.instance).createNavigationPanel();
+			((SimulationScreen) COMPANY_PANEL.instance).createNavigationPanel();
```
Just call initialize. Nothing else.

zcmoore replied:
Navigation panel should be shared among common screens when possible

zcmoore commented on core/src/edu/asu/c3simulator/screens/AllManagementScreens.java:
```diff
@@ -0,0 +1,90 @@
+	 */
+	public Screen getInstance()
+	{
+		System.out.println("instance obtained");
```
Use logger or get rid of this

jreigel committed:
Initializing Screens Fixed Documentation

Initialized council screens and MainHub screen

zcmoore commented on core/src/edu/asu/c3simulator/screens/BusinessCreationGuide.java:
```diff
@@ -18,7 +19,7 @@ public BusinessCreationGuide(Game game)
 		resetAllGuideScreens();
 		//TODO add Industry screen
-		game.setScreen(new IndustryScreen(game));
+		game.setScreen(AllManagementScreens.INDUSTRY.getInstance());
```
Not as clean as I'd like it to be, but better than before. Maybe we can add a method to C3Simulator that takes a ManagementScreens enumeration...

zcmoore replied:
Why does the constructor trigger a redirect?

zcmoore replied:
This looks like a method, not a class...
What does this class actually provide?

jreigel committed:
Removed unnecessary imports, deleted SellCompany

SellCompany was replaced with SellCompanyButton.

zcmoore commented on core/src/edu/asu/c3simulator/screens/BusinessDirectonScreen.java:
```diff
@@ -91,11 +92,9 @@ public void clicked(InputEvent event, float x, float y)
 		
 		// TODO add Corner Advisor
-		// TODO add Home Button
+		HomeButton home = new HomeButton(game);
```
I'd name this homeButton

jreigel replied:
fixed. Github isn't auto removing your previous comment for some reason though.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CompanyPanel.java:
```diff
@@ -214,25 +215,14 @@ private void updateBusinessGraph()
-				- (0.3f * stage.getHeight()));
-		return navigationPanel;
+		@SuppressWarnings(value = { "unused" })
+		CouncilPanel navigation = new CouncilPanel(game, stage);
```
Dead store.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CompletedTasks.java:
```diff
@@ -78,14 +79,11 @@ public CompletedTasks(Game game)
 		
 		updateCompletedTasks();
 		
-		stage.addActor(navigationPanel);
```
Why are we removing the navigation panel? Don't we need it on this screen?

jreigel replied:
It is in the createNavigationPanel() method below.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CompletedTasks.java:
```diff
@@ -78,14 +79,11 @@ public CompletedTasks(Game game)
-		
-		// TODO add screens
-		NavigationPanel navigationPanel = createNavigationPanel();
+		HomeButton home = new HomeButton(game);
```
Rename to homeButton

jreigel replied:
Done.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CompletedTasks.java:
```diff
@@ -139,21 +137,10 @@ public void updateCompletedTasks()
-				- (0.3f * stage.getHeight()));
-		return navigationPanel;
+		@SuppressWarnings(value = { "unused" })
+		CouncilPanel navigation = new CouncilPanel(game, stage);
```
Another dead store. Why are we storing this?

zcmoore commented on core/src/edu/asu/c3simulator/screens/CouncilPanel.java:
```diff
@@ -0,0 +1,46 @@
+import edu.asu.c3simulator.widgets.NavigationPanel;
+
+/**
+ * Navigation Panel for council screens
```
Sounds like this should be a factory method instead of its own class.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CouncilPanel.java:
```diff
@@ -0,0 +1,46 @@
+/**
+ * Navigation Panel for council screens
+ * 
+ * @author Alyahya, Mohammed
```
We need to verify that this class isn't registered in any other branches or commits.

malyahy2 replied:
I can verify that. We created it in Justin's share branch when we were pair programming on linking the screens

jreigel replied:
It isn't. Mohammed and I were pair programming a few days ago and just added all the code in share_branch_justin.

zcmoore replied:
Perfect

zcmoore commented on core/src/edu/asu/c3simulator/screens/CouncilPanel.java:
```diff
@@ -0,0 +1,46 @@
+{
+	private static final int DESIGN_WIDTH = 1280;
+	private static final int DESIGN_HEIGHT = 720;
+	float padding = 0.01f * DESIGN_HEIGHT;
```
Why is this package-protected?

zcmoore commented on core/src/edu/asu/c3simulator/screens/CouncilPanel.java:
```diff
@@ -0,0 +1,46 @@
+	private static final int DESIGN_WIDTH = 1280;
+	private static final int DESIGN_HEIGHT = 720;
+	float padding = 0.01f * DESIGN_HEIGHT;
+	private static NavigationPanel navigation;
```
Is this used?

zcmoore commented on core/src/edu/asu/c3simulator/screens/CouncilPanel.java:
```diff
@@ -0,0 +1,46 @@
+		navigationPanel.addSubButton("Company", "Business",
+				AllManagementScreens.COMPANY_PANEL.getInstance());
+		navigationPanel.addSubButton("Company", "Assets", null);
+		navigationPanel.addButton("Tasks", null);
```
Needs link or TODO

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+import com.badlogic.gdx.utils.viewport.StretchViewport;
+import com.badlogic.gdx.utils.viewport.Viewport;
+
+//import edu.asu.c3simulator.widgets.CornerAdvisor;
```
Commented block

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	private Stage stage;
+	private Skin skin;
+	private Game game;
+	// private CornerAdvisor advisor;
```
Remove commented areas. If needed, add a TODO

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	private Skin skin;
+	private Game game;
+	// private CornerAdvisor advisor;
+	private Table table;
```
Non-descriptive name. What is this for?

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	private Table table;
+	private BitmapFont font = new BitmapFont(
+			Gdx.files.internal("fonts/arial32_superSample.fnt"));
+	private ManagePanel manage;
```
managementPanel would be a better name. It could be improved even more.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+
+		// advisor = new CornerAdvisor(ADVISOR_TEXT);
+		// creates and adds manage Panel to the current screen
+		// manage = new ManagePanel(game);
```
Commented code violates current policy

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+		// float advisorLeft = DESIGN_WIDTH - advisor.getPrefWidth() - padding;
+		// float advisorBottom = DESIGN_HEIGHT - advisor.getPrefHeight() -
+		// padding;
+		// advisor.setPosition(advisorLeft, advisorBottom);
```
Commented block

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+		HomeButton home = new HomeButton(game);
+
+		stage.addActor(home);
+		// stage.addActor(advisor);
```
Commented block - remove

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	public void initialize()
+	{
+		manage = new ManagePanel(game, stage);
+	}
```
You should move all of the **practical** instantiation code here, from the constructor.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	 * 
+	 * @return Table
+	 */
+	private Actor ProductPane()
```
methods should be camelCase starting with lower case

zcmoore replied:
A better name may be createProductPanel()

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+
+	/**
+	 * Creates a table with detailed information about the current products in a
+	 * panel that the can be scrolled through.
```
The documentation says it uses information about the current product, but the functionality does not reflect this. Will this panel be updated, or recreated when the current product changes? (or a new product becomes the current product)

jreigel replied:
added a TODO

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	 * 
+	 * @return Table
+	 */
+	private Actor ProductInfo()
```
Better name would be createProductInformationPanel

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	@Override
+	public void pause()
+	{
+		throw new UnsupportedOperationException(
```
Add a TODO, or implement the actual functionality

If adding a TODO, specify the exact functionality that is desired

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	@Override
+	public void render(float delta)
+	{
+		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
```
Replace with C3Simulator.clearScreen();

zcmoore commented on core/src/edu/asu/c3simulator/screens/CurrentProducts.java:
```diff
@@ -0,0 +1,202 @@
+	{
+		throw new UnsupportedOperationException(
+				"The method is not implemented yet.");
+	}
```
See note on pause()

zcmoore commented on core/src/edu/asu/c3simulator/screens/DifficultySelectionScreen.java:
```diff
@@ -17,9 +17,9 @@
- * to the main hub
+ * Allows the user to select the game difficulty. Each available difficulty is
+ * displayed along with the details of the difficulty. Clicking an option will
+ * transition the player to the main hub
```
Did you decrease your line width?

jreigel replied:
no.

zcmoore commented on core/src/edu/asu/c3simulator/screens/DifficultySelectionScreen.java:
```diff
@@ -185,5 +192,5 @@ public void show()
 	}
-	
+
 }
```
All changes to this class (DifficultySelectionScreen) should be rejected

jreigel replied:
This was just the result of ctrl+shift+f. also, I linked it to mainhub/removed todo comment asking to link it to mainhub.

zcmoore replied:
Related question: Did you decrease your line width?

zcmoore replied:
In the future, don't commit these changes, unless they are committed as a cleanup commit in their own branch/pull.

Also, it looks like your settings diverged at some point. Indented lines are acceptable. Please update your formatter to comply.

jreigel replied:
I'm not sure what your settings are for line width or format, but I am just using the default.

zcmoore commented on core/src/edu/asu/c3simulator/screens/EmploymentScreen.java:
```diff
@@ -50,11 +50,13 @@
 	 * clicked, the selected employee will be set as the selected employee, and their
 	 * information will be displayed in the employee model.
 	 */
+	private ManagePanel manage;
```
Where is this used?

zcmoore commented on core/src/edu/asu/c3simulator/screens/FundingScreen.java:
```diff
@@ -49,29 +50,27 @@ public void clicked(InputEvent event, float x, float y)
 		
 		// TODO add Corner Advisor
-		// TODO add Home Button
+		HomeButton home = new HomeButton(game);
```
Better name: homeButton

zcmoore commented on core/src/edu/asu/c3simulator/screens/FundingScreen.java:
```diff
@@ -49,29 +50,27 @@ public void clicked(InputEvent event, float x, float y)
 	
 	@Override
-	public NavigationPanel createNavigationPanel()
+	public void createNavigationPanel()
```
Why was this changed?

zcmoore replied:
The design patterns for modularized panel creation seems inconsistent.

malyahy2 replied:
The purpose of having this method is to create all navigation panels after all screens are created and referenced. However, while working with Justin on linking the screens, I noticed that I was still creating them in the constructor of each screen. So, we changed to make the methods create and add the panel to their screen

jreigel replied:
NavigationPanel return was no longer needed. Method now adds the navigation panel to the stage.

zcmoore replied:
@malyahy2 The problem then, is that the navigationPanel cannot be created in the constructor, because it requires instances of other screens in order to be created, but the other screens cannot be created before the navigation panel. The solution, was the enumerated singleton, AllManagementScreens, in addition to deferred instantiation, via instantiate() methods. Moving the navigation panel creation to the instantiate() method will solve this problem. Alternatively, the navigationPanel could take an AllManagementScreens instance (e.g., AllManagementScreens.INDUSTRY) as an argument, instead of a Screen, and then direct game to screen.getInstance();

The latter is the cleanest solution proposed so far, followed by the former. However, publicizing createNavigationPanel() and using that method to add to the stage directly adds unnecessary bloat, and creates inconsistencies.

zcmoore replied:
Method should probably be made into a factory method, as it is used across screens.

malyahy2 replied:
Actually, the solutions you proposed are all used in the code. The "createNavigationPanel" has exactly the same functionality as the "initialize" method you and justin created. it is just a different name. Also, all navigation panels take some AllManagementScreens instance. The problem that we encountered in our previous demo was that we tried to reference an AllManagementScreens instance that wasn't initialized yet which for some reason resulted in it being considered null even after initialization.

zcmoore replied:
1: Lines 65-68 look like navigationPanel is accepting a Screen, not an AllManagementScreens instance
e.g.: navigationPanel.addButton("Industry", AllManagementScreens.INDUSTRY.getInstance());
AllManagementScreens.INDUSTRY.getInstance() is of type Screen

2: if createNavigationPanel **has exactly the same functionality as the "initialize" method**, then it should be renamed to initialize in order to be consistent with the rest of the code. 

2.1: AllManagementScreens.initialize() should only need to call the initialize() method of its screens, and nothing else. This will allow us to use a loop instead of hard coding. maintainability++ && size-- == true

malyahy2 replied:
1: Well, the thing is that the addButton method takes a string and a screen as arguments. So, if we want to do that then the NavigationPanel class needs to apply these changes.

2: We did change everything to the createNavigationPanel name since it is the one in the SimulationScreen inerface.

zcmoore replied:
1: This comment was made in reply to **all navigation panels take some AllManagementScreens instance** in the context of **Alternatively, the navigationPanel could take an AllManagementScreens instance (e.g., AllManagementScreens.INDUSTRY) as an argument, instead of a Screen, and then direct game to screen.getInstance();**

2: Most lines were changed, except for MainHub, which has a separate cast, and calls instantiate() (line 67 of AllManagementScreens).

2.1: The idea is to generalize it to the point that we can enumerate all screens, and ensure a single instance of every screen. In order to do that, they all need to follow a single interface, with methods that apply to all screens. I think "initialize" is more generalized than "createNavigationPanel" and has the added benefit of being a common design pattern.

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+import com.badlogic.gdx.utils.viewport.StretchViewport;
+import com.badlogic.gdx.utils.viewport.Viewport;
+
+//import edu.asu.c3simulator.widgets.CornerAdvisor;
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+//import edu.asu.c3simulator.widgets.CornerAdvisor;
+import edu.asu.c3simulator.widgets.NavigationPanel;
+
+//import edu.asu.c3simulator.widgets.PlayerStatusDisplay;
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+public class MainHub implements Screen
+{
+
+	private static final String ADVISOR_TEXT = "This is a test of TextAreaX. This is intended to cover multiple lines at a width of 200px. This is the second extention";
```
Is this used?

jreigel replied:
It was, but I commented out advisor in case we were going to keep it. I commented all code with advisor out and removed the advisor class for the time being.

zcmoore replied:
Let's remove all of the leftovers and scaffolding. It can be re-added if need be in the future.

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+	private static final int DESIGN_WIDTH = 1280;
+	private static final int DESIGN_HEIGHT = 720;
+	private static final int DESIGN_SCREEN_CENTER_X = DESIGN_WIDTH / 2;
+	private static final int DESIGN_SCREEN_CENTER_Y = DESIGN_HEIGHT / 2;
```
Check these for unused private variables

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+	private static final int DESIGN_SCREEN_CENTER_Y = DESIGN_HEIGHT / 2;
+
+	@SuppressWarnings("unused")
+	private Game game;
```
Why is this still unused? Transition functionality seems very important for the main **hub**

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+
+	private Stage stage;
+	private Skin skin;
+	// private CornerAdvisor advisor;
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+
+	// TODO: uncomment advisor when advisor is implemented in master_testing or
+	// higher.
+	// public PlayerStatusDisplay player;
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+			Gdx.files.internal("fonts/arial32_superSample.fnt"));
+
+	// TODO: uncomment advisor when advisor is implemented in master_testing or
+	// higher.
```
Fix line width

jreigel replied:
I am unsure of the exact line width you are using.

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+
+		AllManagementScreens.initialize(game);
+
+		// advisor = new CornerAdvisor(ADVISOR_TEXT, Color.BLACK);
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+		// float advisorLeft = DESIGN_WIDTH - advisor.getPrefWidth() - padding;
+		// float advisorBottom = DESIGN_HEIGHT - advisor.getPrefHeight() -
+		// padding;
+		// advisor.setPosition(advisorLeft, advisorBottom);
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+		stage.addActor(company);
+		stage.addActor(navigation);
+		stage.addActor(Council());
+		// stage.addActor(playerInfo());
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+	 * 
+	 * @return Table
+	 */
+	private Actor Company()
```
Bad method name

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+	{
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException(
+				"The method is not implemented yet.");
```
Fix todo. See other pause() comment

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+	@Override
+	public void render(float delta)
+	{
+		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
```
Replace with C3Simulation.clearScreen()

jreigel replied:
Why? This is how I have always been rendering the screens and it works fine.

zcmoore replied:
1: In the past there wasn't the option to use C3Simulation.clearScreen() - it is relatively new, and helps tie together our code.

2: Yes, line 166 works, however, using C3Simulation.clearScreen() has significant benefits to maintainability, including bug tracking, updating, and debugging. Also, it has readability benefits thrown in for free.

zcmoore commented on core/src/edu/asu/c3simulator/screens/MainHub.java:
```diff
@@ -0,0 +1,223 @@
+	{
+		// TODO Auto-generated method stub
+		throw new UnsupportedOperationException(
+				"The method is not implemented yet.");
```
See pause()

jreigel replied:
added TODO

zcmoore replied:
Cheers.

zcmoore commented on core/src/edu/asu/c3simulator/screens/ManagePanel.java:
```diff
@@ -0,0 +1,54 @@
+import edu.asu.c3simulator.widgets.NavigationPanel;
+
+/**
+ * Creates a panel that allows for navigation through the management screens.
```
Sounds like a method again. I want to know what ManagementPanel IS and what value it adds (what it can do, what data it manages (if applicable)) in this class javadoc

zcmoore commented on core/src/edu/asu/c3simulator/screens/ManagePanel.java:
```diff
@@ -0,0 +1,54 @@
+{
+	private static final int DESIGN_WIDTH = 1280;
+	private static final int DESIGN_HEIGHT = 720;
+	float padding = 0.01f * DESIGN_HEIGHT;
```
Are we accessing these in the package? Will we be in the future? 
This design decision is worth a team discussion

zcmoore commented on core/src/edu/asu/c3simulator/screens/ManagePanel.java:
```diff
@@ -0,0 +1,54 @@
+		navigation.setPosition(0.01f * stage.getWidth(),
+				stage.getHeight() * .7f);
+
+		stage.addActor(navigation);
```
Let's replace this with a  factory method that returns a navigation panel, and does **not** add itself to the stage.

It doesn't look like this class adds anything of value

zcmoore commented on core/src/edu/asu/c3simulator/screens/MarketingScreen.java:
```diff
@@ -141,25 +140,10 @@ public MarketingScreen(Game game)
 	 */
 	@Override
-	public NavigationPanel createNavigationPanel()
+	public void createNavigationPanel()
```
~~Again, why was this changed. Refer to conversation above,~~

zcmoore replied:
Same issue here as line 61 of FundingScreen

malyahy2 replied:
Addressed above.

zcmoore commented on core/src/edu/asu/c3simulator/screens/MarketingScreen.java:
```diff
@@ -141,25 +140,10 @@ public MarketingScreen(Game game)
-		
-		return navigationPanel;
+		@SuppressWarnings("unused")
+		ManagePanel navigation = new ManagePanel(game, stage);
```
Tonight, on 10 o'clock news central, we cover the serial killer that keeps bringing these data stores to their grave. Detective C3 investigates.

zcmoore commented on core/src/edu/asu/c3simulator/screens/PreMarketProducts.java:
```diff
@@ -0,0 +1,214 @@
+import com.badlogic.gdx.utils.viewport.StretchViewport;
+import com.badlogic.gdx.utils.viewport.Viewport;
+
+//import edu.asu.c3simulator.widgets.CornerAdvisor;
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/PreMarketProducts.java:
```diff
@@ -0,0 +1,214 @@
+	private Table table;
+	private BitmapFont font = new BitmapFont(
+			Gdx.files.internal("fonts/arial32_superSample.fnt"));
+	private ManagePanel manage;
```
Check for unused members

zcmoore commented on core/src/edu/asu/c3simulator/screens/PreMarketProducts.java:
```diff
@@ -0,0 +1,214 @@
+
+		// advisor = new CornerAdvisor(ADVISOR_TEXT);
+		// creates and adds manage Panel to the current screen
+		// manage = new ManagePanel(game);
```
Remove commented blocks

zcmoore commented on core/src/edu/asu/c3simulator/screens/PreMarketProducts.java:
```diff
@@ -0,0 +1,214 @@
+		// float advisorLeft = DESIGN_WIDTH - advisor.getPrefWidth() - padding;
+		// float advisorBottom = DESIGN_HEIGHT - advisor.getPrefHeight() -
+		// padding;
+		// advisor.setPosition(advisorLeft, advisorBottom);
```
Remove commented blocks

zcmoore commented on desktop/.gitignore:
```diff
@@ -0,0 +1,2 @@
+/bin
+/bin
```
Another duplicate?

jreigel replied:
Again unsure of why this is happening or what it will affect. Should I just delete them both?

zcmoore replied:
I'd make sure /bin is listed once, and I'd make sure it is being ignored properly, but yes: delete the duplicates.

jreigel replied:
fixed

zcmoore commented on core/testing/edu/asu/c3simulator/testing/stubs/ProductTestingStub.java:
```diff
@@ -0,0 +1,205 @@
+/**
+ * 
+ */
```
Get rid of this

zcmoore commented on core/testing/edu/asu/c3simulator/testing/stubs/ProductTestingStub.java:
```diff
@@ -0,0 +1,205 @@
+	private int efficiency;
+	/* (non-Javadoc)
+	 * @see edu.asu.c3simulator.simulation.Product#getProduct()
+	 */
```
Get rid of these non-javadocs, unless they add something of value (in which case they should be javadoc comments)

zcmoore commented on core/src/edu/asu/c3simulator/widgets/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+ */
+public class SellCompany extends Table
+{
+
```
Extra line. Get rid of this

zcmoore commented on core/src/edu/asu/c3simulator/widgets/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+	private BitmapFont font;
+	private Window window;
+
+	public SellCompany(final Game game, final Skin skin)
```
Any reason these were declared final?

zcmoore commented on core/src/edu/asu/c3simulator/widgets/SellCompany.java:
```diff
@@ -0,0 +1,87 @@
+package edu.asu.c3simulator.widgets;
```
I think I reviewed this above. Get rid of the duplicates, and make sure they are identical.

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductInfoPanel.java:
```diff
@@ -0,0 +1,86 @@
+/**
+ * 
+ */
```
Remove

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductInfoPanel.java:
```diff
@@ -0,0 +1,86 @@
+		Label text3 = new Label("Design employee", skin);
+		productInfo.add(text3).left().row();
+		// TODO: place real employees in dropdown box, add functionality when
+		// selected
```
Fix line width

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductInfoPanel.java:
```diff
@@ -0,0 +1,86 @@
+	/**
+	 * adds product history to the panel
+	 */
+	public void historyPanel()
```
Non-descriptive method name. Javadoc needs work. It is misleading, and needs formatting (capitalization, etc)

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductInfoPanel.java:
```diff
@@ -0,0 +1,86 @@
+		productInfo.add(businessSelectionBox).row();
+		productInfo.add("").row();
+
+		Label text4 = new Label("Materials: $" + testProduct.getMaterialCost(),
```
Variable name should be descriptive.

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductInfoPanel.java:
```diff
@@ -0,0 +1,86 @@
+		Label text = new Label("Production cost: $"
+				+ testProduct.getProductionCost(), skin);
+		productInfo.add(text).left().row();
+		Label text2 = new Label("Labor: ", skin);
```
Was this auto-generated?

zcmoore commented on core/src/edu/asu/c3simulator/widgets/NewProductButton.java:
```diff
@@ -0,0 +1,36 @@
+//			}
+//		});
+//	}
+//}
```
What is this?

zcmoore commented on core/src/edu/asu/c3simulator/widgets/HomeButton.java:
```diff
@@ -0,0 +1,46 @@
+package edu.asu.c3simulator.widgets;
```
~~I think I reviewed this above. Get rid of the duplicates, and make sure they are identical.~~
Resolved.

jreigel replied:
Fixed.

zcmoore commented on core/src/edu/asu/c3simulator/util/Product.java:
```diff
@@ -44,7 +47,7 @@ private Image getImage(String imagePath)
 	public Image getProductImage()
 	{
-		return productImage;
+		return getImage(imagePath);
```
This looks really inefficient. What is the reason for this change? Do we need the image path? Do we need unique copies of the image?

malyahy2 replied:
Yes, we do need unique copies of the image. A lot of screens use the product image as icons and add listeners to them. Having new images returned will make it possible to add different listeners to the same product's icon. For example, screens that manage current product and the marketing screen should all get the same list pf product objects from the company data layer. So they should each create a new icon for user interaction appropriate to their respected screen.

zcmoore replied:
1: **A lot of screens use the product image as icons and add listeners to them** Ah. That makes sense.

2: **Having new images returned will make it possible to add different listeners to the same product's icon** An alternative is to create a new container each time, and add the image to the container. That should reduce load time for the images, and still allow listeners to function individually. However, now that you've explained the reasoning behind it my initial concerns are lessened.

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerImp.java:
```diff
@@ -0,0 +1,89 @@
+import edu.asu.c3simulator.simulation.SocioeconomicStatus;
+
+/**
+ * Implementation of the Player interface
```
Need more detail. Add an @see tag. Describe what this implementation is/does. See documentation of LinkedList vs ArrayList for a high-level example

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerImp.java:
```diff
@@ -0,0 +1,89 @@
+ *
+ */
+
+public class PlayerImp implements Player
```
Naming these classes is always difficult. We should think of a better name than PlayerImp, though.

zcmoore commented on core/src/edu/asu/c3simulator/simulation/SimulationScreen.java:
```diff
@@ -2,21 +2,19 @@
+	 * edited by Justin
 	 */
-	NavigationPanel createNavigationPanel();
+	void createNavigationPanel();
```
~~What was the reason for this change?~~
Same issue here as line 61 of FundingScreen

zcmoore commented on core/src/edu/asu/c3simulator/simulation/Product.java:
```diff
@@ -0,0 +1,35 @@
+	public void setEfficiency();
+	//Change which laborer/s are currently producing the product.
+	public void setLabor();
+	//allows the product to be retired.
```
replace with javadoc

jreigel replied:
Fixed.

zcmoore commented on core/src/edu/asu/c3simulator/simulation/Product.java:
```diff
@@ -0,0 +1,35 @@
+	public void setSellingPrice();
+	public void setEfficiency();
+	//Change which laborer/s are currently producing the product.
+	public void setLabor();
```
Interfaces do not need to specify "public" or "abstract" as all interface methods gain these automatically. 

Also, if Product is just a structure, eliminating the interface may be cleaner. I doubt we will need multiple implementations of product.

jreigel replied:
Fixed.

zcmoore commented on core/src/edu/asu/c3simulator/screens/TasksScreen.java:
```diff
@@ -70,19 +70,19 @@ public void clicked(InputEvent event, float x, float y)
 	
 	@Override
-	public NavigationPanel createNavigationPanel()
+	public void createNavigationPanel()
```
~~See other change comments.~~
Same issue here as line 61 of FundingScreen

zcmoore commented on core/src/edu/asu/c3simulator/screens/TaskManagementScreen.java:
```diff
@@ -29,31 +30,17 @@ public TaskManagementScreen(Game game)
-		
-		return navigationPanel;
+		@SuppressWarnings(value = { "unused" })
+		CouncilPanel navigation = new CouncilPanel(game, stage);
```
It was Colonel Mustard, in the bitstream, with the E-clipse. 

Mystery solved. We can all go home.

zcmoore commented on core/src/edu/asu/c3simulator/screens/TaskManagementScreen.java:
```diff
@@ -29,31 +30,17 @@ public TaskManagementScreen(Game game)
 		
 		// TODO add Corner Advisor
-		// TODO add Home Button
+		HomeButton home = new HomeButton(game);
```
Better name: homeButton. Replace with factory method

zcmoore commented on core/src/edu/asu/c3simulator/screens/RetiredProducts.java:
```diff
@@ -0,0 +1,206 @@
+package edu.asu.c3simulator.screens;
```
This class has a lot of mistakes similar or identical to other cases. I am postponing the review of this class until it is refactored to address the relative issues mentioned elsewhere in this review.

jreigel replied:
Refactored.

zcmoore commented on core/src/edu/asu/c3simulator/screens/ProductHistory.java:
```diff
@@ -0,0 +1,180 @@
+package edu.asu.c3simulator.screens;
```
This class has a lot of mistakes similar or identical to other cases. I am postponing the review of this class until it is refactored to address the relative issues mentioned elsewhere in this review.

jreigel replied:
Refactored.

zcmoore commented on core/src/edu/asu/c3simulator/screens/PreMarketProducts.java:
```diff
@@ -0,0 +1,214 @@
+	{
+		Gdx.input.setInputProcessor(stage);
+	}
+}
```
This class has a lot of mistakes similar or identical to other cases. I am postponing the remainder of the review of this class until it is refactored to address the relative issues mentioned elsewhere in this review.

jreigel replied:
Refactored.

zcmoore commented:
Most of this commit was flagged as potential copy&paste, and I see it as well. Most cases can be refactored and reduced through modules, inheritance, and interfacing. I suggest you run the c&p check, and evaluate the results. I also mentioned the highlights in this review.

Additionally, this pull is riddled with small but abundant issues. As such, it was difficult to offer a critical review of some portions. Once the redundant issues are addressed, I recommend this pull for a second review.

zcmoore commented on core/src/edu/asu/c3simulator/screens/CompanyPanel.java:
```diff
@@ -240,45 +229,45 @@ public void render(float delta)
 	}
-	
+
 }
```
These spacing changes make reviewing difficult. If you want to format the spacing of files, please make a new pull on a new branch, and request direct entry to master

zcmoore replied:
I just realized that these spacing changes are reversed. Don't remove indents in the future.

jreigel committed:
Refactored branch

Fixed documentation, removed duplicates, removed unused fields, removed
unneeded comments, removed unneeded classes.

jreigel committed:
Deleted PlayerImp class

jreigel committed:
BusinessDirectionScreen Product

Changed homeButton varaible name in BusinessDirectionScreen  Fixed
comments in Product.

jreigel committed:
CompanyPanel SellCompanyButton

Added SellCompanyButton to companyPanel. Improved SellCompanyButton
popups.

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerInterfaceImplementation.java:
```diff
@@ -0,0 +1,89 @@
+ *
+ */
+
+public class PlayerInterfaceImplementation implements Player
```
I'll mention this again: naming schemes in these instances (where an interface with the "perfect" name is being used, and we need an implementation class for it) are difficult to devise, and are a problem we should tackle as a group, or at least open a gitHub discussion for it. PlayerInterfaceImplementation, PlayerImp, etc. are generally smurfy, and I think we can do better if we collectively put our heads together. 

Once again, ArrayList and LinkedList both implement List, yet they have meaningful names that depict their interface **and** their unique aspects. HashMap, HashTable, and LinkedHashMap are other good Java examples of good naming conventions. I don't have a good proposal for this case (Player), but I'd like to open a group conversation to figure it out. PlayerInterfaceImplementation is bulky and doesn't add much.

For now, do the following so we can get this into master_testing:
- **Open an issue on GitHub to discuss the naming topic**
- Add a TODO tag on line 14. 
- Add an @version tag in the javadoc (I'd go with V0.1)
- Update the javadoc to reflect the fact that this class will be deprecated and replaced with a refactored version in the near future.

zcmoore replied:
The comment below about this being the live data layer presents another requirement

jreigel replied:
Updated and added TODOs for the live data layer.

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerInterfaceImplementation.java:
```diff
@@ -0,0 +1,89 @@
+
+	/**
+	 * @return The net worth of the player in base units
+	 */
```
This is good, as it specifies units.

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerInterfaceImplementation.java:
```diff
@@ -0,0 +1,89 @@
+	private C3Simulation simulation;
+
+	public PlayerInterfaceImplementation(SocioeconomicStatus newStatus, int newNetWorth,
+			int newCapital, C3Simulation newSimulation)
```
Avoid names like: "newStatus" "newBleh" "new..." etc. Instead, use the name of the data member (e.g., status, netWorth, etc) and specify the class-scope by using this.status, this.netWorth, etc.

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerInterfaceImplementation.java:
```diff
@@ -0,0 +1,89 @@
+		netWorth = newNetWorth;
+		capital = newCapital;
+		simulation = newSimulation;
+	}
```
I just realized that this is the live data layer, not a stub...

We need to address the following issues (and possible more) and decide if the player is responsible for these or if another class will take care of them, and if so, how:
- When/how is netWorth updated? 
  - I think calculating it based on live data would be more robust and less prone to defects, but there may be a better way.
- When/how is status updated? 
- How is status determined?
- When/how is capital updated?
- How do I get a list of companies the player owns?
- Does the player have any other features, such as a line of credit or assets?
- Is it practical to allow players to exist in different simulations? 
  - I think simulation may be singleton, static, or similar
- Is the player class just a structure, or is it responsible for certain calculations or activities?

jreigel replied:
Honestly, I can delete it if needed and make a new one in sprint 4. It is not used anywhere. It was used in MainHub, but then I never got around to answering any of those questions and removed it from MainHub.

jreigel replied:
Added TODO

zcmoore replied:
Was it used in MainHub or not?

zcmoore commented on core/src/edu/asu/c3simulator/util/PlayerInterfaceImplementation.java:
```diff
@@ -0,0 +1,89 @@
+	/**
+	 * @return Parent simulation to which this player belongs
+	 */
+	@Override
```
Just a note: our styleguide does not require javadocs for methods with the @Override annotation, but they are acceptable if they add something new.

Also note: If these comments are identical to the comments in the interface that they override, then they should be removed. If a method has an @Override annotation but no javadoc, it reverts to its parent's documentation (which makes maintaining the documentation much easier).

Copy + Paste = Bad +- 0.05

zcmoore commented on core/src/edu/asu/c3simulator/widgets/HomeButton.java:
```diff
@@ -0,0 +1,45 @@
+				game.setScreen(AllManagementScreens.MAIN_HUB.getInstance());
+			}
+		});
+	}
```
Better as a factory method

malyahy2 replied:
why?
I really don't understand why is one better than the other.

zcmoore replied:
Classes add overhead, and are bulky. Generally, if a class is going to be instantiated, we should get some value from the expense of having the class (i.e. it should provide some functionality, or data relationship that isn't achieved otherwise). 

In this case, the class doesn't actually do anything, except assemble a table.

zcmoore replied:
Using a factory method instead of a class will improve maintainability, reduce our LOC, and provide better performance.

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductDesigns.java:
```diff
@@ -109,7 +110,7 @@ public void clicked(InputEvent event, float x, float y)
 			}
 		});
-		
+		newProductPopupWindow.clear();
```
Fantastic

jreigel committed:
.gitignore

Removed extra /bin lines from gitignore

zcmoore commented on core/src/edu/asu/c3simulator/widgets/ProductInfoPanel.java:
```diff
@@ -0,0 +1,83 @@
+
+/**
+ * Creates a panel with detailed product info for product screens in the manage
+ * section of the hub.
```
Is this class a builder?

zcmoore commented on core/src/edu/asu/c3simulator/widgets/SellCompanyButton.java:
```diff
@@ -0,0 +1,101 @@
+			}
+		});
+
+	}
```
This looks like a factory method to me

zcmoore commented on core/src/edu/asu/c3simulator/screens/IndustryScreen.java:
```diff
@@ -49,29 +50,27 @@ public void clicked(InputEvent event, float x, float y)
 	
 	@Override
-	public NavigationPanel createNavigationPanel()
+	public void createNavigationPanel()
```
Same issue here as line 61 of FundingScreen

malyahy2 committed:
Added a Factory for Navigation panel & Formatting

jreigel committed:
PlayerInterfaceImplementaion

updated documentation,  put in TODO lines where needed.

jreigel committed:
PlayerInterfaceImplementation

Fixed format.
